client/server/render/index.js

```diff
@@ -233,6 +233,7 @@ export function serverRender( req, res ) {
 		attachHead( context );
 
 		const cacheableReduxSubtrees = [ 'documentHead' ];
+		const cacheableLoggedOutReduxSubtrees = ! context.isLoggedIn ? [ 'themes', 'ui' ] : [];
```

Comment:
Also tried `context.user` but it wasn't returning the correct result (ie. false when logged in). 
`context.isLoggedIn` worked well, as long as the instance is running on `wordpres.com` domain.

Reply:
`context.user` should be fine when we're using the bootstrapped user (e.g. for staging and production). See https://github.com/Automattic/wp-calypso/blob/0d17396107f113bd675fb37ff0e0559817975045/client/server/pages/index.js#L302

Reply:
> `context.user` should be fine when we're using the bootstrapped user (e.g. for staging and production). See

I stand corrected. It appears there're a few more steps to getting the bootstrapped user on development. Retested with `context.user` and worked as expected.